azero/contracts/advisory/contract/lib.rs

```diff
@@ -0,0 +1,79 @@
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+#[ink::contract]
+pub mod advisory {
+
+    use advisory_trait::{AdvisoryError, IsAdvisory};
+
+    #[ink(event)]
+    #[derive(Debug)]
+    #[cfg_attr(feature = "std", derive(Eq, PartialEq))]
+    pub struct EmergencyChanged {
+        pub emergency: bool,
+    }
+
+    #[ink(event)]
+    #[derive(Debug)]
+    #[cfg_attr(feature = "std", derive(Eq, PartialEq))]
+    pub struct OwnershipTransferred {
+        pub previous_owner: AccountId,
+        pub new_owner: AccountId,
+    }
+
+    #[ink(storage)]
+    pub struct Advisory {
+        pub owner: AccountId,
+        pub emergency: bool,
+    }
+
+    impl IsAdvisory for Advisory {
+        #[ink(message)]
+        fn flip_emergency(&mut self) -> Result<(), AdvisoryError> {
+            self.ensure_owner()?;
+            self.emergency = !self.emergency;
+            self.env().emit_event(EmergencyChanged {
+                emergency: self.emergency,
+            });
+            Ok(())
+        }
+
```

Comment:
Suggestion: perhaps it would be cleaner if this were a `set_emergency` instead? Given multiple advisories exist, it would be unfortunate if someone sets or unsets this by accident, for example, because they think they are interacting with another one?

Reply:
Good idea, done